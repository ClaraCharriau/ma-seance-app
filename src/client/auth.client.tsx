import { User } from '../models/User';
import axios from 'axios';

const user_auth_path = 'http://localhost:7878/auth';
const user_verify_path = 'http://localhost:7878/verify';
const user_sign_in_path = 'http://localhost:7878/sign-in';

export const loginUser = async (email: string, password: string): Promise<User> => {
    return await axios
        .post(user_auth_path, {
            email: email,
            password: password
        })
        .then(response => {
            console.log('user-auth-client-response POST user :', response.data);
            return response.data;
        })
        .catch(error => {
            console.error('Error: could not log user : ' + error.message);
            throw error;
        });
};

export const checkAccountExists = async (email: string): Promise<boolean> => {
    return await axios
        .post(user_verify_path, {
            email: email
        })
        .then(function (response) {
            console.log('user-verify-client-response POST verify :', response.data);
            if (response.data && typeof response.data.exists === 'boolean') {
                return response.data.exists;
            }
            throw Error('Could not parse api /verify response. ');
        })
        .catch(function (error: Error) {
            console.error('Error: could not verify if user exists' + error.message);
            throw error;
        });
};

export const signIn = async (pseudo: string, email: string, password: string): Promise<User> => {
    return await axios
        .post(user_sign_in_path, {
            pseudo: pseudo,
            email: email,
            password: password
        })
        .then(function (response) {
            console.log('user-sign-in-client-response POST user :', response.data);
            return response.data;
        })
        .catch(function (error: Error) {
            console.error('Error: could not sign in user' + error.message);
            throw error;
        });
};
