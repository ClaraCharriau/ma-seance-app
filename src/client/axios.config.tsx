import axios, { AxiosResponse } from 'axios';
import { toast } from 'react-toastify';
import config from '../config/config.helper';
import { UserToken } from '../model/UserToken';

/**
 * Set axios global configuration
 */
export const axiosInstance = axios.create({
    baseURL: config.services.prefix,
    headers: {
        'Content-Type': 'application/json'
    }
});

/**
 * Set authentication header if token is found
 */
axiosInstance.interceptors.request.use(config => {
    const maSeanceIdToken = localStorage.getItem('maSeanceId');
    if (maSeanceIdToken) {
        const parsedUsertoken: UserToken = JSON.parse(maSeanceIdToken);
        config.headers.Authorization = `Bearer ${parsedUsertoken.access_token}`;
    }
    return config;
});

/**
 * Set global behavior for responses
 */
axiosInstance.interceptors.response.use(
    response => handleResponse(response),
    error => handleError(error)
);

export const handleError = (error: any) => {
    if (error.response.status === 401 || error.response.status === 403) {
        throw new Response(error.response.data, {
            status: error.response.status,
            statusText: error.response.statusText
        });
    }
    toast.error("Une erreur s'est produite.");
    if (error.response) {
        throw new Response(error.response.data, {
            status: error.response.status,
            statusText: error.response.statusText
        });
    }
};

export const handleResponse = (response: AxiosResponse) => {
    return response.data;
};
