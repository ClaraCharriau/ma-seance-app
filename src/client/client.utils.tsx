import { AxiosResponse } from 'axios';

export const HOST = 'http://localhost:7878';

export const handleError = (error: any) => {
    if (error.response) {
        throw new Response(error.response.data, {
            status: error.response.status,
            statusText: error.response.statusText
        });
    }
    console.error(error.message);
    throw error;
};

export const handleResponse = (response: AxiosResponse) => {
    return response.data;
};

export const handleDeleteResponse = (response: AxiosResponse) => {
    return response;
};
