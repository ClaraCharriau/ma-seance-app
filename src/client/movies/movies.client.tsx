import { Movie } from '../../models/Movie';
import { TheaterScreenings } from '../../models/TheaterScreenings';
import { axiosInstance } from '../axios.config';

const MOVIES_PATH = '/movies';
const CURRENTLY_PATH = MOVIES_PATH + '/currently';
const SCREENINGS_PATH = '/screenings';

export const getCurrentlyMovies = async (): Promise<Movie[]> => {
    return await axiosInstance.get(CURRENTLY_PATH);
};

export const getMovieById = async (movieId: string): Promise<Movie> => {
    return await axiosInstance.get(MOVIES_PATH + '/' + movieId);
};

export const getTheaterScreeningsByMovieIdAndDay = async (
    movieId: string,
    day: string
): Promise<TheaterScreenings[]> => {
    const params = {
        day
    };
    return await axiosInstance.get(MOVIES_PATH + '/' + movieId + SCREENINGS_PATH, { params });
};
