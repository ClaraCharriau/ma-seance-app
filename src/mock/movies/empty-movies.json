{
    "_metadata": {
        "page": 5,
        "per_page": 20,
        "page_count": 20,
        "total_count": 521
    },
    "records": []
}
