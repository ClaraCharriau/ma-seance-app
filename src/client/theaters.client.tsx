import axios from 'axios';
import { HOST, handleError, handleResponse } from './client.utils';
import { Theater } from '../models/Theater';

const theaters_path = HOST + '/theaters';

export const getTheaterById = async (theaterId: number): Promise<Theater> => {
    return await axios
        .get(theaters_path + '/' + theaterId)
        .then(response => handleResponse(response))
        .catch(error => handleError(error));
};
