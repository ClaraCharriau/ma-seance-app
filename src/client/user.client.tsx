import { Theater } from '../models/Theater';
import axios from 'axios';
import { handleError, handleResponse } from './client.utils';

const host = 'http://localhost:7878';
const users_path = host + '/users';

export const getUserFavTheaters = async (id: number): Promise<Theater[]> => {
    return await axios
        .get(users_path + '/' + id + '/fav-theaters')
        .then(response => handleResponse(response))
        .catch(error => handleError(error, 'GET /fav-theaters'));
};
