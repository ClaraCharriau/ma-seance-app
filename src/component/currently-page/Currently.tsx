import { useEffect, useState } from 'react';
import MovieList from '../common/movie-list/MovieList';
import { getCurrentlyMovies } from '../../client/movies/movies.client';
import Spinner from '../common/spinner/Spinner';
import { Movie } from '../../models/Movie';
import useCurrentTheaterWeek from '../../hooks/useCurrentWeek';
import style from './Currently.module.css';

const Currently = () => {
    const [currentMovies, setCurrentMovies] = useState<Movie[]>([]);
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState('');
    const ERROR_MESSAGE = "Une erreur s'est produite lors du chargement des films.";
    const currentWeek = useCurrentTheaterWeek();

    useEffect(() => {
        const timeoutId = setTimeout(() => {
            setError(ERROR_MESSAGE);
            setLoading(false);
        }, 5000); // 5s

        getCurrentlyMovies()
            .then(movies => {
                setCurrentMovies(movies);
                setLoading(false);
                clearTimeout(timeoutId);
            })
            .catch(() => {
                setError(ERROR_MESSAGE);
                setLoading(false);
            });

        return () => clearTimeout(timeoutId);
        // eslint-disable-next-line
    }, []);

    return (
        <main>
            <h2 className={style.pageTitle}>
                <svg width="24" height="25" viewBox="0 0 24 25" fill="none" xmlns="http://www.w3.org/2000/svg">
                    <path
                        d="M20.3797 13.5808C13.2146 12.2051 12.2359 11.2264 10.861 4.06129C10.8398 3.95091 10.7807 3.85138 10.6941 3.7798C10.6074 3.70822 10.4986 3.66907 10.3862 3.66907C10.2738 3.66907 10.1649 3.70822 10.0782 3.7798C9.99158 3.85138 9.93257 3.95091 9.91135 4.06129C8.53646 11.2264 7.55772 12.2051 0.392208 13.5808C0.281863 13.6019 0.182333 13.6609 0.110748 13.7475C0.0391627 13.8341 0 13.9429 0 14.0552C0 14.1676 0.0391627 14.2764 0.110748 14.363C0.182333 14.4496 0.281863 14.5085 0.392208 14.5297C7.55769 15.9049 8.53646 16.8829 9.91135 24.0488C9.93267 24.1591 9.99172 24.2585 10.0784 24.33C10.165 24.4015 10.2738 24.4406 10.3862 24.4406C10.4985 24.4406 10.6073 24.4015 10.694 24.33C10.7806 24.2585 10.8397 24.1591 10.861 24.0488C12.2359 16.8829 13.2146 15.9049 20.3797 14.5297C20.4901 14.5085 20.5896 14.4496 20.6612 14.363C20.7328 14.2764 20.7719 14.1676 20.7719 14.0552C20.7719 13.9429 20.7328 13.8341 20.6612 13.7475C20.5896 13.6609 20.4901 13.6019 20.3797 13.5808Z"
                        fill="#FFF1A7"
                    />
                    <path
                        d="M19.8474 8.30516C19.735 8.30511 19.6261 8.26593 19.5395 8.19436C19.4529 8.1228 19.3938 8.0233 19.3726 7.91296C18.8883 5.38821 18.6118 5.1117 16.087 4.62741C15.9766 4.60618 15.8771 4.54716 15.8055 4.46051C15.734 4.37386 15.6948 4.26498 15.6948 4.15259C15.6948 4.0402 15.734 3.93132 15.8055 3.84467C15.8771 3.75802 15.9766 3.69901 16.087 3.67778C18.6122 3.19312 18.8883 2.91657 19.3726 0.392219C19.3938 0.281846 19.4528 0.182315 19.5395 0.110735C19.6261 0.0391555 19.735 0 19.8474 0C19.9598 0 20.0687 0.0391555 20.1553 0.110735C20.242 0.182315 20.301 0.281846 20.3222 0.392219C20.8065 2.91657 21.0826 3.19312 23.6078 3.67778C23.7181 3.69901 23.8176 3.75802 23.8892 3.84467C23.9608 3.93132 23.9999 4.0402 23.9999 4.15259C23.9999 4.26498 23.9608 4.37386 23.8892 4.46051C23.8176 4.54716 23.7181 4.60618 23.6078 4.62741C21.083 5.11166 20.8065 5.38821 20.3222 7.91296C20.3009 8.0233 20.2419 8.1228 20.1553 8.19437C20.0686 8.26594 19.9598 8.30511 19.8474 8.30516ZM18.0988 4.15259C18.5069 4.28641 18.8779 4.51438 19.1816 4.81807C19.4854 5.12177 19.7135 5.49266 19.8474 5.90079C19.9813 5.49266 20.2094 5.12177 20.5131 4.81807C20.8169 4.51438 21.1878 4.28641 21.596 4.15259C21.1879 4.01864 20.817 3.79063 20.5133 3.48695C20.2095 3.18328 19.9814 2.81245 19.8474 2.40438C19.7133 2.81244 19.4852 3.18328 19.1815 3.48695C18.8778 3.79063 18.5069 4.01864 18.0988 4.15259Z"
                        fill="#FFF1A7"
                    />
                </svg>
                Sorties de la semaine
            </h2>
            <p className={style.currentWeek}>{currentWeek}</p>
            {loading ? <Spinner /> : error ? <div>{error}</div> : <MovieList movieList={currentMovies} />}
        </main>
    );
};

export default Currently;
