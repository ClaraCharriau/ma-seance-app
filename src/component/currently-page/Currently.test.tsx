import { BrowserRouter } from 'react-router-dom';
import Currently from './Currently';
import { render, waitFor } from '@testing-library/react';
import { getCurrentlyMovies } from '../../client/movies/movies.client';
import mockMovies from '../../mocks/movies/current-movies.json';

jest.mock('../../client/movies/movies.client', () => ({
    getCurrentlyMovies: jest.fn()
}));
const mockGetCurrentlyMovies = getCurrentlyMovies as jest.MockedFunction<typeof getCurrentlyMovies>;

describe('Currently component test', () => {
    jest.useFakeTimers().setSystemTime(new Date('2020-01-01'));
    it('should render component with a list of current movies', async () => {
        // Given
        mockGetCurrentlyMovies.mockResolvedValue(mockMovies);

        // When
        const component = render(
            <BrowserRouter>
                <Currently />
            </BrowserRouter>
        );

        // Then
        await waitFor(() => {
            expect(component.getByText("Madame Web")).toBeInTheDocument();
        });
        expect(component.baseElement).toMatchSnapshot();
    });

    it('should fail to render a list of current movies', async () => {
        // Given
        mockGetCurrentlyMovies.mockRejectedValue(new Error());

        // When
        const component = render(
            <BrowserRouter>
                <Currently />
            </BrowserRouter>
        );

        // Then
        await waitFor(() => {
            expect(component.getByText("Une erreur s'est produite lors du chargement des films.")).toBeInTheDocument();
            expect(component.container).toMatchSnapshot();
        });
    });
});
