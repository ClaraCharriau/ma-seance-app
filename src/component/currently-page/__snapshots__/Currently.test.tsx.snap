// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currently component test should fail to render a list of current movies 1`] = `
<div>
  <main>
    <h2
      class="pageTitle"
    >
      <svg
        fill="none"
        height="25"
        viewBox="0 0 24 25"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.3797 13.5808C13.2146 12.2051 12.2359 11.2264 10.861 4.06129C10.8398 3.95091 10.7807 3.85138 10.6941 3.7798C10.6074 3.70822 10.4986 3.66907 10.3862 3.66907C10.2738 3.66907 10.1649 3.70822 10.0782 3.7798C9.99158 3.85138 9.93257 3.95091 9.91135 4.06129C8.53646 11.2264 7.55772 12.2051 0.392208 13.5808C0.281863 13.6019 0.182333 13.6609 0.110748 13.7475C0.0391627 13.8341 0 13.9429 0 14.0552C0 14.1676 0.0391627 14.2764 0.110748 14.363C0.182333 14.4496 0.281863 14.5085 0.392208 14.5297C7.55769 15.9049 8.53646 16.8829 9.91135 24.0488C9.93267 24.1591 9.99172 24.2585 10.0784 24.33C10.165 24.4015 10.2738 24.4406 10.3862 24.4406C10.4985 24.4406 10.6073 24.4015 10.694 24.33C10.7806 24.2585 10.8397 24.1591 10.861 24.0488C12.2359 16.8829 13.2146 15.9049 20.3797 14.5297C20.4901 14.5085 20.5896 14.4496 20.6612 14.363C20.7328 14.2764 20.7719 14.1676 20.7719 14.0552C20.7719 13.9429 20.7328 13.8341 20.6612 13.7475C20.5896 13.6609 20.4901 13.6019 20.3797 13.5808Z"
          fill="#FFF1A7"
        />
        <path
          d="M19.8474 8.30516C19.735 8.30511 19.6261 8.26593 19.5395 8.19436C19.4529 8.1228 19.3938 8.0233 19.3726 7.91296C18.8883 5.38821 18.6118 5.1117 16.087 4.62741C15.9766 4.60618 15.8771 4.54716 15.8055 4.46051C15.734 4.37386 15.6948 4.26498 15.6948 4.15259C15.6948 4.0402 15.734 3.93132 15.8055 3.84467C15.8771 3.75802 15.9766 3.69901 16.087 3.67778C18.6122 3.19312 18.8883 2.91657 19.3726 0.392219C19.3938 0.281846 19.4528 0.182315 19.5395 0.110735C19.6261 0.0391555 19.735 0 19.8474 0C19.9598 0 20.0687 0.0391555 20.1553 0.110735C20.242 0.182315 20.301 0.281846 20.3222 0.392219C20.8065 2.91657 21.0826 3.19312 23.6078 3.67778C23.7181 3.69901 23.8176 3.75802 23.8892 3.84467C23.9608 3.93132 23.9999 4.0402 23.9999 4.15259C23.9999 4.26498 23.9608 4.37386 23.8892 4.46051C23.8176 4.54716 23.7181 4.60618 23.6078 4.62741C21.083 5.11166 20.8065 5.38821 20.3222 7.91296C20.3009 8.0233 20.2419 8.1228 20.1553 8.19437C20.0686 8.26594 19.9598 8.30511 19.8474 8.30516ZM18.0988 4.15259C18.5069 4.28641 18.8779 4.51438 19.1816 4.81807C19.4854 5.12177 19.7135 5.49266 19.8474 5.90079C19.9813 5.49266 20.2094 5.12177 20.5131 4.81807C20.8169 4.51438 21.1878 4.28641 21.596 4.15259C21.1879 4.01864 20.817 3.79063 20.5133 3.48695C20.2095 3.18328 19.9814 2.81245 19.8474 2.40438C19.7133 2.81244 19.4852 3.18328 19.1815 3.48695C18.8778 3.79063 18.5069 4.01864 18.0988 4.15259Z"
          fill="#FFF1A7"
        />
      </svg>
      Sorties de la semaine
    </h2>
    <p
      class="currentWeek"
    >
      Semaine du 1 au 14 janvier 2020
    </p>
    <div>
      Une erreur s'est produite lors du chargement des films.
    </div>
  </main>
</div>
`;

exports[`Currently component test should render component with a list of current movies 1`] = `
<body>
  <div>
    <main>
      <h2
        class="pageTitle"
      >
        <svg
          fill="none"
          height="25"
          viewBox="0 0 24 25"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.3797 13.5808C13.2146 12.2051 12.2359 11.2264 10.861 4.06129C10.8398 3.95091 10.7807 3.85138 10.6941 3.7798C10.6074 3.70822 10.4986 3.66907 10.3862 3.66907C10.2738 3.66907 10.1649 3.70822 10.0782 3.7798C9.99158 3.85138 9.93257 3.95091 9.91135 4.06129C8.53646 11.2264 7.55772 12.2051 0.392208 13.5808C0.281863 13.6019 0.182333 13.6609 0.110748 13.7475C0.0391627 13.8341 0 13.9429 0 14.0552C0 14.1676 0.0391627 14.2764 0.110748 14.363C0.182333 14.4496 0.281863 14.5085 0.392208 14.5297C7.55769 15.9049 8.53646 16.8829 9.91135 24.0488C9.93267 24.1591 9.99172 24.2585 10.0784 24.33C10.165 24.4015 10.2738 24.4406 10.3862 24.4406C10.4985 24.4406 10.6073 24.4015 10.694 24.33C10.7806 24.2585 10.8397 24.1591 10.861 24.0488C12.2359 16.8829 13.2146 15.9049 20.3797 14.5297C20.4901 14.5085 20.5896 14.4496 20.6612 14.363C20.7328 14.2764 20.7719 14.1676 20.7719 14.0552C20.7719 13.9429 20.7328 13.8341 20.6612 13.7475C20.5896 13.6609 20.4901 13.6019 20.3797 13.5808Z"
            fill="#FFF1A7"
          />
          <path
            d="M19.8474 8.30516C19.735 8.30511 19.6261 8.26593 19.5395 8.19436C19.4529 8.1228 19.3938 8.0233 19.3726 7.91296C18.8883 5.38821 18.6118 5.1117 16.087 4.62741C15.9766 4.60618 15.8771 4.54716 15.8055 4.46051C15.734 4.37386 15.6948 4.26498 15.6948 4.15259C15.6948 4.0402 15.734 3.93132 15.8055 3.84467C15.8771 3.75802 15.9766 3.69901 16.087 3.67778C18.6122 3.19312 18.8883 2.91657 19.3726 0.392219C19.3938 0.281846 19.4528 0.182315 19.5395 0.110735C19.6261 0.0391555 19.735 0 19.8474 0C19.9598 0 20.0687 0.0391555 20.1553 0.110735C20.242 0.182315 20.301 0.281846 20.3222 0.392219C20.8065 2.91657 21.0826 3.19312 23.6078 3.67778C23.7181 3.69901 23.8176 3.75802 23.8892 3.84467C23.9608 3.93132 23.9999 4.0402 23.9999 4.15259C23.9999 4.26498 23.9608 4.37386 23.8892 4.46051C23.8176 4.54716 23.7181 4.60618 23.6078 4.62741C21.083 5.11166 20.8065 5.38821 20.3222 7.91296C20.3009 8.0233 20.2419 8.1228 20.1553 8.19437C20.0686 8.26594 19.9598 8.30511 19.8474 8.30516ZM18.0988 4.15259C18.5069 4.28641 18.8779 4.51438 19.1816 4.81807C19.4854 5.12177 19.7135 5.49266 19.8474 5.90079C19.9813 5.49266 20.2094 5.12177 20.5131 4.81807C20.8169 4.51438 21.1878 4.28641 21.596 4.15259C21.1879 4.01864 20.817 3.79063 20.5133 3.48695C20.2095 3.18328 19.9814 2.81245 19.8474 2.40438C19.7133 2.81244 19.4852 3.18328 19.1815 3.48695C18.8778 3.79063 18.5069 4.01864 18.0988 4.15259Z"
            fill="#FFF1A7"
          />
        </svg>
        Sorties de la semaine
      </h2>
      <p
        class="currentWeek"
      >
        Semaine du 1 au 14 janvier 2020
      </p>
      <section
        class="movieList"
      >
        <div
          class="movieListCard"
        >
          <a
            class="movieCard"
            href="/movies/3e07de0f-f2c8-4c46-9ac9-744abb8af481/day-1"
          >
            <img
              alt="affiche du film Your Name."
              class="movieCardImg"
              src="https://image.tmdb.org/t/p/w500/zyHjvVRgKOt9wgVx4ikp2kGArGF.jpg"
            />
            <div>
              <p
                class="movieTitle"
              >
                Your Name.
              </p>
              <p
                class="movieDateDuration"
              >
                2016
                 - 
                1h47
              </p>
              <p
                class="movieDirector"
              >
                Réalisé par 
                Makoto Shinkai
              </p>
              <p
                class="movieCast"
              >
                Avec 
                Ryunosuke Kamiki, Mone Kamishiraishi, Ryo Narita
              </p>
            </div>
          </a>
        </div>
        <div
          class="movieListCard"
        >
          <a
            class="movieCard"
            href="/movies/3d8f1342-15f1-44b1-a48f-4581d654b94a/day-1"
          >
            <img
              alt="affiche du film Pauvres créatures"
              class="movieCardImg"
              src="https://image.tmdb.org/t/p/w500/5fT98da9ccWN2xr8VOJrSBp3Cdw.jpg"
            />
            <div>
              <p
                class="movieTitle"
              >
                Pauvres créatures
              </p>
              <p
                class="movieDateDuration"
              >
                2024
                 - 
                1h30
              </p>
              <p
                class="movieDirector"
              >
                Réalisé par 
                Yorgos Lanthimos
              </p>
              <p
                class="movieCast"
              >
                Avec 
                Emma Stone, Mark Ruffalo, Willem Dafoe
              </p>
            </div>
          </a>
        </div>
        <div
          class="movieListCard"
        >
          <a
            class="movieCard"
            href="/movies/f7e41334-179d-451a-a1d4-e1b9a60ee785/day-1"
          >
            <img
              alt="affiche du film Argylle"
              class="movieCardImg"
              src="https://image.tmdb.org/t/p/w500/uVAk2YliqInQfH4B4vzZ75rwcNB.jpg"
            />
            <div>
              <p
                class="movieTitle"
              >
                Argylle
              </p>
              <p
                class="movieDateDuration"
              >
                2024
                 - 
                1h30
              </p>
              <p
                class="movieDirector"
              >
                Réalisé par 
                Matthew Vaughn
              </p>
              <p
                class="movieCast"
              >
                Avec 
                Henry Cavill, Sam Rockwell, Dua Lipa
              </p>
            </div>
          </a>
        </div>
        <div
          class="movieListCard"
        >
          <a
            class="movieCard"
            href="/movies/51e755d5-ec43-4e77-a2e6-f4f82a0e8ec5/day-1"
          >
            <img
              alt="affiche du film Night Swim"
              class="movieCardImg"
              src="https://image.tmdb.org/t/p/w500/1njKOrYQPQKAAX0Wt3PFptX4ULG.jpg"
            />
            <div>
              <p
                class="movieTitle"
              >
                Night Swim
              </p>
              <p
                class="movieDateDuration"
              >
                2024
                 - 
                1h30
              </p>
              <p
                class="movieDirector"
              >
                Réalisé par 
                Bryce McGuire
              </p>
              <p
                class="movieCast"
              >
                Avec 
                Wyatt Russell, Kerry Condon, Gavin Warren
              </p>
            </div>
          </a>
        </div>
        <div
          class="movieListCard"
        >
          <a
            class="movieCard"
            href="/movies/b88f9510-d302-47d5-9d6b-8b13740f541d/day-1"
          >
            <img
              alt="affiche du film Madame Web"
              class="movieCardImg"
              src="https://image.tmdb.org/t/p/w500/rULWuutDcN5NvtiZi4FRPzRYWSh.jpg"
            />
            <div>
              <p
                class="movieTitle"
              >
                Madame Web
              </p>
              <p
                class="movieDateDuration"
              >
                2024
                 - 
                1h30
              </p>
              <p
                class="movieDirector"
              >
                Réalisé par 
                S. J. Clarkson
              </p>
              <p
                class="movieCast"
              >
                Avec 
                Dakota Johnson, Sydney Sweeney, Isabela Merced
              </p>
            </div>
          </a>
        </div>
        <div
          class="movieListCard"
        >
          <a
            class="movieCard"
            href="/movies/e301d3db-6aaf-47b6-a003-6a1bb0a3f029/day-1"
          >
            <img
              alt="affiche du film La Couleur Pourpre"
              class="movieCardImg"
              src="https://image.tmdb.org/t/p/w500/fKkWAjAwNXHHtxnpufRUbgFpmxr.jpg"
            />
            <div>
              <p
                class="movieTitle"
              >
                La Couleur Pourpre
              </p>
              <p
                class="movieDateDuration"
              >
                2024
                 - 
                1h30
              </p>
              <p
                class="movieDirector"
              >
                Réalisé par 
                Blitz Bazawule
              </p>
              <p
                class="movieCast"
              >
                Avec 
                Fantasia Barrino & araji P. Henson
              </p>
            </div>
          </a>
        </div>
        <div
          class="movieListCard"
        >
          <a
            class="movieCard"
            href="/movies/b1d0de57-b205-4927-a13e-85e9405f5d8f/day-1"
          >
            <img
              alt="affiche du film Mean Girls - Lolita Malgré Moi"
              class="movieCardImg"
              src="https://image.tmdb.org/t/p/w500/fVuPnZAEsF4ssCvUuQxISKzdDTt.jpg"
            />
            <div>
              <p
                class="movieTitle"
              >
                Mean Girls - Lolita Malgré Moi
              </p>
              <p
                class="movieDateDuration"
              >
                2024
                 - 
                1h30
              </p>
              <p
                class="movieDirector"
              >
                Réalisé par 
                Samantha Jayne & Arturo Perez Jr
              </p>
              <p
                class="movieCast"
              >
                Avec 
                Angourie Rice & Reneé Rapp
              </p>
            </div>
          </a>
        </div>
        <div
          class="movieListCard"
        >
          <a
            class="movieCard"
            href="/movies/03a20bb3-b24b-4d76-af62-0b0fe4d03b91/day-1"
          >
            <img
              alt="affiche du film Bob Marley : One Love"
              class="movieCardImg"
              src="https://image.tmdb.org/t/p/w500/iN6WZnetgrf2gnS4bMdCDZLDkar.jpg"
            />
            <div>
              <p
                class="movieTitle"
              >
                Bob Marley : One Love
              </p>
              <p
                class="movieDateDuration"
              >
                2024
                 - 
                1h30
              </p>
              <p
                class="movieDirector"
              >
                Réalisé par 
                Reinaldo Marcus Green
              </p>
              <p
                class="movieCast"
              >
                Avec 
                Kingsley Ben-Adir, Lashana Lynch, Michael Gandolfini
              </p>
            </div>
          </a>
        </div>
      </section>
    </main>
  </div>
</body>
`;
